flink-dist/src/main/flink-bin/bin/flink

```diff
@@ -17,6 +17,20 @@
 # limitations under the License.
 ################################################################################
 
+function abspath() {
+    if [ -d "$1" ]; then
+        # dir
+        (cd "$1"; pwd)
+    elif [ -f "$1" ]; then
+        # file
+        if [[ $1 == */* ]]; then
+            echo "$(cd "${1%/*}"; pwd)/${1##*/}"
+        else
+            echo "$(pwd)/$1"
```

Comment:
Have you looked in `config.sh`? There is a function for resolving symbolic links. What happens in case of cyclic symbolic links?